### PR TITLE
Add exists option

### LINKED ISSUE #5: Match to bool
**Is your feature request related to a problem? Please describe.**
It would be nice if there was a way store whether there was a match into a boolean.

**Describe the solution you'd like**
I was thinking there could be an additional struct tag to configure this but I am not sure what it should look like. Booleans are currently supported but only to match true or false.

```go
type B struct {
	HasSuffixFoo bool `regroup:"foo,flag"`
}

var re = regroup.MustCompile(`.*(?P<foo>foo)?`)
```

**Describe alternatives you've considered**
Currently I just check the value of the struct after matching.

### PR DESCRIPTION
**Pull request by eberkund**
Closes #5 

How does this look @oriser?

### CODE REVIEW

**Review by oriser — COMMENTED** [review]
@eberkund That's look great!
Thank you for your contribution!
The tests CI is failing, looks like it because the go version of the CI needs to be updated as well.
Can you please update it here?
https://github.com/oriser/regroup/blob/fca8d753126329fe3e12fa1d88325eb25803b498/.github/workflows/go.yml#L11

Also, please add a small demo and documentation to the README file, so others will know how to use your feature :)

**Comment by eberkund** [comment]
I just updated both, can you try running the CI workflow again?

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-go@v2
-        name: Setup go
+      - uses: actions/setup-go@v4
+        name: Setup Go
         with:
-          go-version: 1.14
-      - uses: actions/checkout@v2
+          go-version-file: "go.mod"
```
Comment (oriser) [inline .github/workflows/go.yml:13]: If you use the go-version-file, the checkout step must be before it

Comment (eberkund) [reply .github/workflows/go.yml:13]: Fixed

**Review by oriser — APPROVED** [review]
Awesome! 😁

**Pull request merged**